docs/arcadia-users-group/20230214-code-review-for-pubs/lesson.md

```diff
@@ -0,0 +1,88 @@
+It's also super helpful for other people to see what you ran so they can use it as a recipe for a reasonable way to analyze their data -- think of it like a tutorial in disguise.
+For figures, we think filtering steps or transformations are important to capture.
+Even if you don't do any of these steps, your figure code is also like a tutorial in disguise – since we provide our input data and pubs openly, providing the code use to make a plot allows other people to interface with our results or to accomplish something similar to what we did.
+We get that coding a pub-ready figure is a big task – it's totally fine to edit titles, labels, and fonts in PowerPoint/Photoshop after making the plot (just be careful not to edit the data!).
+
+Imagine a world in which you pick up a manuscript that has some really interesting new result.
+Wouldn't it be cool if you could you could access the code for the figures inline, and recreate them all, but zooming in on the results you specifically care about?
+Or even cooler – what if you could follow the exact recipe documented in the manuscript for your data, and then quickly add your observations to the plots in the original manuscript?
+[Executable manuscripts](https://www.nature.com/articles/s42005-020-00403-4) are a thing, but they're hard for reasons beyond code and data availability. 
+Even still, providing the code we use – including for our figures – is both a key component to achieving these products and a suitable minimum replacement when the full thing isn't possible.
+
+**Question:** Could code review happen after version 1 of the pub?
+
+**Thoughts:** The goal of code review in the pub release process is to make sure someone inside or outside of Arcadia could pick up the code you used and with minimal effort, reproduce the results you got and reuse the code you wrote.
+We think that code review prior to the release of the first version of a pub will increase the usefulness of the computational products we're releasing to the world.
+
+Our goal is to have the code review component of the pub release process become a formality; we want code review to happen quickly after a unit of code is written (a new function, a new analysis notebook, a new figure) so that the feedback you get is actually useful and doesn't slow you down.
+If we can get to that point, the checks in the pub release cycle will be super quick.
+To make this process faster and to reduce the heat around version 1 pub releases, we've created [a checklist](#minimums-for-passing-code-review-a-checklist) of the five things that are needed to approve code associated with the first release of a pub.
+Any other suggestions you get are just that – suggestions – and can be punted to later pub releases or ignored.
+
+## Overview of the pub process and how it relates to code review
+
+![](overview.png)
+
+Checkout the **What to expect from code review during the pub process** Notion page for an up to date guide through this process.
```

Comment:
link

Reply:
Since this is public, I didn't link it...is it safe to link it and then people without access to our notion just won't be able to see it?

Reply:
yeah it's safe

Reply:
```suggestion
Checkout the [What to expect from code review during the pub process](https://www.notion.so/arcadiascience/What-to-expect-from-code-review-during-the-pub-process-eac7ab5083ff4d88bf3039686b9281b7) Notion page for an up to date guide through this process.
```